src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,124 @@
+package duke;
+
+import duke.command.AddDeadlineCommand;
+import duke.command.AddEventCommand;
+import duke.command.AddTodoCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.FinishCommand;
+import duke.command.ListCommand;
+
+public class Parser {
+
+    public static class ParsedCommand {
+        public final String input;
+        public final String keyword;
+        public final String args;
+        public final String beforeSlashArgs;
+        public final String slashKeyword;
+        public final String slashArgs;
+
+        /**
+         * Creates a parsed command, whereby the input command as a string is parsed into its components.
+         * Components include the entire trimmed input, keyword, arguments, arguments before the slash keyword,
+         * arguments after the slash keyword, and the slash keyword itself.
+         * @param input Input command as a string
+         */
+        public ParsedCommand(String input) {
+            input = input.trim();
+            this.input = input;
+
+            this.keyword = input.split("\\s+", 2)[0];
+            this.args = input.split("\\s+").length >= 2 ? input.split("\\s+", 2)[1] : null;
+
+            String inputBeforeSlash = input.split("\\s+/", 2)[0];
+            String inputAfterSlash = input.split("\\s+/").length >= 2 ? input.split("\\s+/", 2)[1] : null;
+
+            this.beforeSlashArgs = inputBeforeSlash.split("\\s+").length >= 2
+                    ? inputBeforeSlash.split("\\s+", 2)[1]
+                    : null;
+
+            if (inputAfterSlash == null) {
+                this.slashKeyword = null;
+                this.slashArgs = null;
+            } else {
+                this.slashKeyword = inputAfterSlash.split("\\s+", 2)[0];
+                this.slashArgs = inputAfterSlash.split("\\s+").length >= 2
+                        ? inputAfterSlash.split("\\s+", 2)[1]
+                        : null;
+            }
+        }
+    }
+
+    private static void validateIsPositiveInteger(String s, String message) throws DukeException {
```

Comment:
Maybe this could be in the UI since it is interpreting the content of the input from users to see if there is any errors? I am not very sure but I think this is very minor :)